### PR TITLE
Issue-3: Add initial support for Tuple

### LINKED ISSUE #3: Add initial support for Tuple
Must include the following:

* A `Tuple` type.
* The following static constructors:
    * `Of :: a → b → (a, b)`
* The following static methods:
    * `First :: (a, b) → a`: Extracts the first component of the pair.
    * `Second :: (a, b) → b`: Extracts the second component of the pair.
    * `Swap :: (a, b) → (b, a)`: Swaps the components of the pair.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #3
SemVer:Minor
enhancement

### CODE REVIEW

**Comment by jlmorgan** [comment]
- [ ]: Squash commits before merging.

File: src/FunctionalExtras/Data/ValueTupleExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+using System;
```
Comment (doomchild) [inline src/FunctionalExtras/Data/ValueTupleExtensions.cs:1]: I think this file is misnamed.  It's dealing with `System.Tuple` instances, not `System.ValueTuple` instances.

File: src/FunctionalExtras/Data/TupleExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+using System;
```
Comment (doomchild) [inline src/FunctionalExtras/Data/TupleExtensions.cs:1]: I think this file is misnamed.  It's dealing with `System.ValueTuple` instances, not `System.Tuple` instances.

File: src/FunctionalExtras/Data/Tuples.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+using System;
```
Comment (doomchild) [inline src/FunctionalExtras/Data/Tuples.cs]: I think this file should be `ValueTuples.cs`.

**Review by doomchild — CHANGES_REQUESTED** [review]

File: src/FunctionalExtras/Data/ValueTupleExtensions.cs (hunk 1 above)
Comment (jlmorgan) [reply src/FunctionalExtras/Data/ValueTupleExtensions.cs:1]: I think it didn't switch context when I clicked… = \

Comment (jlmorgan) [reply src/FunctionalExtras/Data/ValueTupleExtensions.cs:1]: Fixed in [783e8ab](https://github.com/jlmorgan/csharp-functional-extras/pull/11/commits/783e8ab2605bb4e45e3098d36592f6c6b90d819b).

File: src/FunctionalExtras.Tests/Data/TuplesTests.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+using System;
```
Comment (doomchild) [inline src/FunctionalExtras.Tests/Data/TuplesTests.cs]: It looks like this is testing `ValueTuple` instances, not `Tuple` instances.

Comment (jlmorgan) [reply src/FunctionalExtras.Tests/Data/TuplesTests.cs]: Oop. Forgot to rename the file… One moment…

Comment (jlmorgan) [reply src/FunctionalExtras.Tests/Data/TuplesTests.cs]: Fixed in `e68fded`.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Pull request merged**